osu.Game/Overlays/Direct/DirectRulesetSelector.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.Direct
+{
+    public class DirectRulesetSelector : RulesetSelector
+    {
+        public override bool HandleNonPositionalInput => !Current.Disabled && base.HandleNonPositionalInput;
+
+        public override bool HandlePositionalInput => !Current.Disabled && base.HandlePositionalInput;
+
+        public override bool PropagatePositionalInputSubTree => !Current.Disabled && base.PropagatePositionalInputSubTree;
+
+        public DirectRulesetSelector()
+        {
+            TabContainer.Masking = false;
+            TabContainer.Spacing = new Vector2(10, 0);
+            AutoSizeAxes = Axes.Both;
+
+            Current.DisabledChanged += value => SelectedTab.FadeColour(value ? Color4.DarkGray : Color4.White, 200, Easing.OutQuint);
+        }
+
+        protected override TabItem<RulesetInfo> CreateTabItem(RulesetInfo value) => new DirectRulesetTabItem(value);
+
+        protected override TabFillFlowContainer CreateTabFlow() => new TabFillFlowContainer
+        {
+            Direction = FillDirection.Horizontal,
+            AutoSizeAxes = Axes.Both,
+        };
+
+        private class DirectRulesetTabItem : TabItem<RulesetInfo>
+        {
+            private readonly SpriteIcon icon;
+
+            public DirectRulesetTabItem(RulesetInfo value)
+                : base(value)
+            {
+                AutoSizeAxes = Axes.Both;
+
+                Children = new Drawable[]
+                {
+                    icon = (SpriteIcon)value.CreateInstance().CreateIcon(),
```

Comment:
It's unsafe to assume this is always going to be an icon. Accept the drawable and use `Scale` (or the `ConstrainedIcon` class like in `ConstrainedIconContainer`) instead if you need to adjust its size.